src/object/bonus_block.cpp

```diff
@@ -266,7 +266,7 @@ BonusBlock::try_open(Player* player)
     return;
 
   if (player == nullptr)
-    player = &Sector::get().get_player();
+    player = Sector::get().get_nearest_player(m_col.m_bbox);
```

Comment:
`get_nearest_player` is bound to cause some strange interactions in some cases, but it's good enough for now, I guess.

Reply:
It picks the nearest player if and only if there is no player that performed the `try_open` (such as an iceblock or a portable, if I look at the source code).

Perhaps an idea would be to make iceblocks/portables open bonus blocks from the perspective of the player who threw the object? But would require having them keep track of which player(s) interact with them, and it wouldn't take into account the cases where they hit a bonus block without any player interaction.